lib/elixir/src/elixir_bitstring.erl

```diff
@@ -255,9 +255,11 @@ validate_spec_arg(Meta, unit, Value, E) when not is_integer(Value) ->
 validate_spec_arg(_Meta, _Key, _Value, _E) ->
   ok.
 
-validate_size_required(Meta, true, default, Type, default, E) when Type == binary; Type == bitstring ->
+validate_size_required(Meta, true, _, _, _, {'^', _, _}, E) ->
+  form_error(Meta, ?key(E, file), ?MODULE, right_side_pin_operator);
```

Comment:
I think we should call the error atom `pin_operator_on_left_side` or something that describes that happened. Wdyt?

Reply:
Makes sense to me, but the error message should still suggest the correct behaviour, right?

Reply:
Oh yes, definitely. Sorry I wasn't clear. I was only referring to the atom `right_side_pin_operator`, the message is fine like it is IMO :).